pkg/controller/ingress/concurrency_watchdog.go

```diff
@@ -24,10 +25,57 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/Azure/aks-app-routing-operator/pkg/config"
-	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
 	"github.com/Azure/aks-app-routing-operator/pkg/util"
 )
 
+// ScrapeFn returns the connection count for the given pod
+type ScrapeFn func(ctx context.Context, client rest.Interface, pod *corev1.Pod) (float64, error)
+
+// NginxScrapeFn is the scrape function for Nginx
+func NginxScrapeFn(ctx context.Context, client rest.Interface, pod *corev1.Pod) (float64, error) {
```

Comment:
follow up on prev comment but why separate this from nginx config? this would only ever be used for an nginx ingress controller right?

Reply:
This was never connected to the nginx config. This is a different abstraction. This package has no concept of the nginx config, just LabelGetters.